integrations/DCMFloodlight/utils.js

```diff
@@ -72,4 +91,224 @@ const mapFlagValue = (key, value) => {
+  if (salesTag) {
+    // sales tag
+    dcCustomParams.ord =
+      get(message, "properties.orderId") || get(message, "properties.order_id");
+    let qty = get(message, "properties.quantity");
+
+    eventSnippetPayload = {
+      ...eventSnippetPayload,
+      value: get(message, "properties.revenue"),
+      transaction_id: dcCustomParams.ord,
+    };
+
+    // sums quantity from products array or fallback to properties.quantity
+    const products = get(message, "properties.products");
+    const quantities = calculateQuantity(products);
+    if (quantities) {
+      qty = quantities;
+    }
+
+    // Ref - https://support.google.com/campaignmanager/answer/7554821#zippy=%2Cfields-in-event-snippets-for-sales-tags
+    switch (countingMethod) {
+      case "transactions":
+        break;
+      case "items_sold":
+        if (qty) {
+          eventSnippetPayload.quantity = parseFloat(qty);
+        }
+        break;
+      default:
+        logger.error("[DCM Floodlight] Sales Tag:: invalid counting method");
+        return;
+    }
+  } else {
+    // counter tag
+    // Ref - https://support.google.com/campaignmanager/answer/7554821#zippy=%2Cfields-in-event-snippets-for-counter-tags
+    switch (countingMethod) {
+      case "standard":
+        break;
+      case "unique":
+        dcCustomParams.num = get(message, "properties.num");
+        break;
+      case "per_session":
+        dcCustomParams.ord = get(message, "properties.sessionId");
+        eventSnippetPayload.session_id = dcCustomParams.ord;
+        break;
+      default:
+        logger.error("[DCM Floodlight] Counter Tag:: invalid counting method");
+        return;
+    }
+  }
+
+  // COPPA, GDPR, npa must be provided inside integration object
+  let { DCM_FLOODLIGHT } = message.integrations;
+  if (!DCM_FLOODLIGHT) {
+    ({ DCM_FLOODLIGHT } = integrationObj);
+  }
+
+  if (DCM_FLOODLIGHT) {
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.COPPA)) {
+      dcCustomParams.tag_for_child_directed_treatment = mapFlagValue(
+        "COPPA",
+        DCM_FLOODLIGHT.COPPA
+      );
+    }
+
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.GDPR)) {
+      dcCustomParams.tfua = mapFlagValue("GDPR", DCM_FLOODLIGHT.GDPR);
+    }
+
+    if (isDefinedAndNotNull(DCM_FLOODLIGHT.npa)) {
+      dcCustomParams.npa = mapFlagValue("npa", DCM_FLOODLIGHT.npa);
+    }
+  }
+
+  if (isDefinedAndNotNull(dcCustomParams.dc_lat)) {
+    dcCustomParams.dc_lat = mapFlagValue("dc_lat", dcCustomParams.dc_lat);
+  }
+
+  const matchId = get(message, "properties.matchId");
+  if (matchId) {
+    dcCustomParams.match_id = matchId;
+  }
+
+  dcCustomParams = removeUndefinedAndNullValues(dcCustomParams);
+  if (Object.keys(dcCustomParams).length > 0) {
+    eventSnippetPayload.dc_custom_params = dcCustomParams;
+  }
+  return eventSnippetPayload;
+};
+
+const buildIframeTrackPayload = (
+  message,
+  salesTag,
+  countingMethod,
+  integrationObj
+) => {
+  const randomNum = Math.random() * 10000000000000;
```

Comment:
Can we use library to generate this?

Reply:
This is provided by DCM . We can check for a library to generate it.